2.4/root/usr/share/container-scripts/httpd/README.md

```diff
@@ -20,57 +20,146 @@ These can range from server-side programming language support to authentication
 Virtual hosting allows one Apache installation to serve many different Web sites."
 
 
-Usage
------
+Usage in OpenShift
+---------------------
+For this, we will assume that you are using the `rhel8/httpd-24` image, available via `httpd:24` imagestream tag in Openshift.
+Building a simple [httpd-sample-app](https://github.com/sclorg/httpd-ex.git) application
+in Openshift can be achieved with the following step:
 
-For this, we will assume that you are using the Apache HTTP Server 2.4 container image from the
-Red Hat Container Catalog called `rhscl/httpd-24-rhel7` available via `httpd:2.4` imagestream tag in Openshift.
-The image can be used as a base image for other applications based on Apache HTTP web server.
+```
+oc new-app httpd:24~https://github.com/sclorg/httpd-ex.git
+```
 
-An example of the data on the host for both the examples above, that will be served by
-Apache HTTP web server:
+**Accessing the application:**
+```
+$ oc get pods
+$ oc exec <pod> -- curl 127.0.0.1:8080
+```
+
+Source-to-Image framework and scripts
+---------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/3.11/creating_images/s2i.html)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+that makes it easy to write images that take application source code as
+an input, use a builder image like this Node.js container image, and produce
+a new image that runs the assembled application as output.
+
+In order to support the Source-to-Image framework, there are some important scripts inside the builder image:
+
+* The `/usr/libexec/s2i/run` script is set as a default command in the resulting container image (the new image with the application artifacts).
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run in order to produce a new image with the application artifacts. The script takes sources of a given application and places them into appropriate directories inside the image. The structure of httpd-app can look like this:
+
+**`./httpd-cfg`**  
+       Can contain additional Apache configuration files (`*.conf`)
+
+**`./httpd-pre-init`**  
+       Can contain shell scripts (`*.sh`) that are sourced before `httpd` is started
+
+**`./httpd-ssl`**  
+       Can contain own SSL certificate (in `certs/` subdirectory) and key (in `private/` subdirectory)
+
+**`./`**  
```

Comment:
Can this be indented? It seems it belongs only to the second bullet - "...The structure of httpd-app can look like this:"